internal/engine/upper.go

```diff
@@ -497,4 +498,22 @@ func (u Upper) reapOldWatchBuilds(ctx context.Context, manifests []model.Manifes
 	return nil
 }
 
+func replayBuildLog(ctx context.Context, state store.EngineState, resourceNumber int) {
```

Comment:
it seems weird for this to be an action, because it doesn't really modify state? i think this is fine for now, but this probably should be a subscriber with some way to get a read lock on the state (or mutable state)